### PR TITLE
Spi slave enablement

### PR DESCRIPTION
**Pull request by NikhitaR-IFX**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

CONTRIBUTING.md also tells you what to expect in the PR process.

Description:
SPISLave enabled with following functions:
- Constructor
- deinit()
- write()
- read()
- write_readinto()

- Hardware tests extended for SPI module
- Documents corrected with latest changes
- Pin connections for HIL test setup updated
- SoftSPI feature discarded as it is not working as expected.
- Removed old spi tests just calling the constructor.

Other not on ports/psoc6/spi code changes:
- Updates in .github/workflows/ports_psoc6.yml to add i2c and spi tests.
- Changes in automatic pin generation script to allow NC as pin as needed by CYHAL API.
- Moved all tests requiring a "stub" to hw_ext/multi_stub.

@jaenrig-ifx @IFX-Anusha @ramya-subramanyam : This PR is now ready to review.

### CODE REVIEW

File: .github/workflows/ports_psoc6.yml [hunk 1]
```diff
@@ -79,7 +79,18 @@ jobs:
         echo " >> PSoC6 single board tests"
         ./psoc6/run_psoc6_tests.sh -c --psoc6 --dev0 ${devs[0]} 
         echo " >> PSoC6 extended hardware tests"
-        ./psoc6/run_psoc6_tests.sh -c --psoc6-hwext --dev0 ${devs0[0]} 
+        ./psoc6/run_psoc6_tests.sh -c --psoc6-hwext --dev0 ${devs0[0]}
+        echo " >> PSoC6 i2c tests"
+        if [ "${{ matrix.board }}" == "CY8CPROTO-062-4343W" ]; then
```
Comment (jaenrig-ifx) [inline .github/workflows/ports_psoc6.yml:84]: By the way, this dirty selection approach (sorry for that 😞)  should be tackle in the rework of this run_psoc6_tests.sh in future. I might have the chance to start as pre-1.0.0.

File: docs/psoc6/quickref.rst [hunk 2]
```diff
@@ -599,28 +558,74 @@ SPI object is created with default settings or settings of previous initializati
 ::
     
     from machine import SPI
-    spi = SPI(0, sck='P11_2', mosi='P11_0', miso='P11_1') # Default assignment: id=0, SCK=P11_2  ,MOSI=P11_0, MISO=P11_1
-    spi.init()
+    spi = SPI(baudrate=1000000, polarity=0,phase=0,bits=8,firstbit=SPI.MSB,sck='P6_2', mosi='P6_0', miso='P6_1')
```
Comment (jaenrig-ifx) [inline docs/psoc6/quickref.rst:561]: Add spaces between arguments for readabilty and consistency: 

```suggestion
    spi = SPI(baudrate=1000000, polarity=0, phase=0, bits=8, firstbit=SPI. MSB, sck='P6_2', mosi='P6_0', miso='P6_1')
```

Probably, the suggestion make the commit msg check fail. But first time I try this in Github.

File: docs/psoc6/quickref.rst [hunk 3]
```diff
@@ -599,28 +558,74 @@ SPI object is created with default settings or settings of previous initializati
...
+      - *phase* can be 0 or 1. Default set to 0.
+      - *bits* is width in bits for each transfer. Only 8 is supported.
+      - *firstbit* can be SPI.MSB or SPI.LSB. Default is SPI.MSB.
+
+    Example:
+        ::
+            
+            from machine import SPISlave
+            
+            spi_slave = spi = SPI(baudrate=1000000, polarity=0,phase=0,bits=8,firstbit=SPI.MSB,ssel="P6_3",sck='P6_2', mosi='P6_0', miso='P6_1')
```
Comment (jaenrig-ifx) [inline docs/psoc6/quickref.rst:600]: same:

```suggestion
            spi_slave = spi = SPI(baudrate=1000000, polarity=0, phase=0, bits=8, firstbit=SPI.MSB, ssel="P6_3", sck='P6_2', mosi='P6_0', miso='P6_1')
```

File: ports/psoc6/machine_spi.c [hunk 4]
```diff
@@ -65,97 +119,70 @@ static inline void spi_obj_free(machine_spi_obj_t *spi_obj_ptr) {
...
+    mp_printf(print, "SPISlave(baudrate=%u, polarity=%u, phase=%u, bits=%u, firstbit=%u, ssel=%d, sck=%d, mosi=%d, miso=%d)",
+        self->baudrate, self->polarity,
         self->phase, self->bits, self->firstbit,
-        self->sck->addr, self->mosi->addr, self->miso->addr);
+        self->ssel->addr, self->sck->addr, self->mosi->addr, self->miso->addr);
 }
 
-mp_obj_t machine_spi_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *all_args) {
-    enum { ARG_id, ARG_baudrate, ARG_polarity, ARG_phase, ARG_bits, ARG_firstbit, ARG_sck, ARG_mosi, ARG_miso };
+mp_obj_t machine_spi_master_init_helper(machine_spi_obj_t *self, size_t n_args, size_t n_kw, const mp_obj_t *all_args) {
```
Comment (jaenrig-ifx) [inline ports/psoc6/machine_spi.c:183]: Mmm, do we really need to machine_spi_xxxx_init_helpers()? We had one only before, no?
Can we just not have one? and if the user pass the ssel for master complain, and if he does not pass it for slave also complain?
Other than that parameter, the rest is the same or?

**Review by IFX-Anusha — APPROVED** [review]

**Comment by jaenrig-ifx** [comment]
@ramya-subramanyam we are going to merge to bring latest HIL changes to main. You can review later.

**Pull request merged**